### PR TITLE
base-system-busybox+runit-void: Using relative symbolic link to for /sbin/init

### PR DESCRIPTION
**Pull request by moul**
Fixing "initrd + busybox' switch_root" based boots

### CODE REVIEW

**Comment by Gottox** [comment]
please increase the revision of the package :+1:

**Comment by moul** [comment]
@Gottox done + rebased upstream/master

**Comment by Gottox** [comment]
Some other opinions?

I don't like to merge stuff that can break boot without a second review.

**Comment by pullmoll** [comment]
I would like to understand the reasoning for this change. It doesn't seem necessary at the first glance.

**Comment by moul** [comment]
/sbin/init is often the entry point when working outside of the rootfs (using a chroot or in my case, from an initrd), testing existence of file using test -x, or test -f, or even readlink will fail to resolve the absolute symlink, and I need to check it before `pivot_root`int inside the rootfs

I saw this issue before in other init systems:

* [systemd](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=750360)
* [upstart](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=590744)
* [ueventd (android)](https://gitlab.com/pbeeler/system_core/commit/33680b106dcb7025de0c37340869e32db61ad190)

**Comment by Gottox** [comment]
It seems reasonable to me. /sbin/init links are broken if you build your own early userspace.

**Comment by moul** [comment]
a rebase is needed

**Comment by moul** [comment]
rebased

File: srcpkgs/base-system-busybox/template [hunk 1]
```diff
@@ -59,5 +59,5 @@ do_install() {
 	mv ${DESTDIR}/usr/sbin/* ${DESTDIR}/usr/bin
 	rm -f ${DESTDIR}/{,s}bin
 	# Provide /sbin/init symlink to runit.
-	ln -s /usr/bin/runit-init ${DESTDIR}/usr/bin/init
+	ln -s ./runit-init ${DESTDIR}/usr/bin/init
```
Comment (xtraeme) [inline srcpkgs/base-system-busybox/template]: You can drop `./` completely.

Comment (moul) [inline srcpkgs/base-system-busybox/template]: done

**Pull request merged**